src/Mutator/Number/DecrementInteger.php

```diff
@@ -38,37 +38,76 @@ protected function mutatesNode(Node $node): bool
             return false;
         }
 
-        return !$this->isZeroComparedWithCountResult($node);
+        return $this->isAllowedComparison($node);
     }
 
-    private function isZeroComparedWithCountResult(Node $node): bool
+    private function isAllowedComparison(Node\Scalar\LNumber $node): bool
     {
         if ($node->value !== 0) {
-            return false;
+            return true;
         }
 
         $parentNode = $node->getAttribute(ParentConnectorVisitor::PARENT_KEY);
 
-        if (!$parentNode instanceof Node\Expr\BinaryOp\Identical
-            && !$parentNode instanceof Node\Expr\BinaryOp\NotIdentical
-            && !$parentNode instanceof Node\Expr\BinaryOp\Equal
-            && !$parentNode instanceof Node\Expr\BinaryOp\NotEqual
-            && !$parentNode instanceof Node\Expr\BinaryOp\Greater
-            && !$parentNode instanceof Node\Expr\BinaryOp\GreaterOrEqual) {
-            return false;
+        if (!$this->isComparison($parentNode)) {
+            return true;
+        }
+
+        if ($parentNode->left instanceof Node\Expr\FuncCall
+            && \in_array(
+                $this->getLowercaseMethodName($parentNode, 'left'),
+                self::COUNT_NAME,
+                true)
+        ) {
+            return $this->isAllowedLeftSideCountComparison($parentNode);
+        }
+
+        if ($parentNode->right instanceof Node\Expr\FuncCall
+            && \in_array(
+                $this->getLowercaseMethodName($parentNode, 'right'),
```

Comment:
I get this method was before, but how is it better than just `$node->right->name->toLowerString();`... That's going to be much easier to understand tbh.